main/unbound/unbound.initd

```diff
@@ -22,36 +23,20 @@ depend() {
 	after auth-dns
 }
 
-checkconfig() {
-	checkpath --directory /var/run/unbound
-	UNBOUND_PIDFILE=$("${UNBOUND_CHECKCONF}" -o pidfile "${UNBOUND_CONFFILE}")
-	return $?
-}
-
 configtest() {
-	ebegin "Checking ${SVCNAME} configuration"
-	checkconfig
-	eend $?
-}
-
-start() {
-	checkconfig || return $?
-	ebegin "Starting ${SVCNAME}"
-	start-stop-daemon --start --pidfile "${UNBOUND_PIDFILE}" \
-		--exec "${UNBOUND_BINARY}" -- -c "${UNBOUND_CONFFILE}"
+	ebegin "Checking $RC_SVCNAME configuration"
+	"$checkcfg" "$cfgfile" 1>/dev/null || return $?
 	eend $?
 }
 
-stop() {
-	checkconfig || return $?
-	ebegin "Stopping ${SVCNAME}"
-	start-stop-daemon --stop --pidfile "${UNBOUND_PIDFILE}"
-	eend $?
+start_pre() {
+	checkpath --directory /run/unbound
+	configtest || return $?
```

Comment:
This is obviously non-sense. `|| return $?` after `configtest` is needed only in `reload()`, not in `start_pre`, for the exact reason @Shizmob said.